src/lib/db/access-store.ts

```diff
@@ -160,4 +166,33 @@ export class AccessStore {
             })
             .delete();
     }
+
+    async getRootRolesForAllUsers(): Promise<IRoleAndUser[]> {
+        const rows = await this.db
+            .select('id', 'name', 'user_id')
+            .from(`${T.ROLES} AS r`)
+            .leftJoin(`${T.ROLE_USER} AS ru`, 'r.id', 'ru.role_id')
+            .where('r.type', '=', 'root');
+
+        return rows.map(row => ({
+            id: +row.id,
+            name: row.name,
+            userId: +row.user_id,
```

Comment:
`+row.id` and `+row.user_id` ? Coercing to number?

Reply:
It's the "Unary Operator" to convert a string to a number. While concise and effective, this isn't a very well-known feature of JavaScript, so it's not advised to use since it may make your code not as easy to understand.